client/src/components/Register.js

```diff
@@ -39,21 +38,21 @@ class Register extends React.Component {
     }
   }
 
-  handlePasswordChange(ev) {
-    this.setState({ password: ev.target.value });
-  }
-
-  handleEmailChange(ev) {
-    this.setState({ email: ev.target.value });
-  }
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
 
-  handleVolunteerChange(ev) {
-    this.setState({ isMusician: ev.target.checked });
+    this.setState({
+      [name]: value
+    });
   }
 
   handleFormSubmit(ev) {
     ev.preventDefault();
-    this.props.registerRequest(this.state.email, this.state.password, this.state.isMusician);
+    const { email, password, isMusician, agreeTerms } = this.state;
```

Comment:
Does it make sense to have these inputs be uncontrolled instead of using state?

Reply:
Hi Nathan!  The render of the Register component depends on agreeTerms and isMusician for disabling the submit button and showing different content.  I find it easier to manage those relationships when the inputs are controlled.  Let me know if you think that's overkill. 😀

Reply:
Cool no worries. LGTM